instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/coroutines/TestUtils.kt

```diff
@@ -68,10 +71,42 @@ suspend fun MapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes: List<Na
     }
 }
 
+suspend fun MapboxNavigation.setNavigationRoutesAndAwaitError(
+    routes: List<NavigationRoute>,
+    legIndex: Int
+) = withTimeout(MAX_TIME_TO_UPDATE_ROUTE) {
+    suspendCoroutine<Unit?> { continuation ->
+        val callback = RoutesSetCallback {
+            if (it.isError) {
+                continuation.resume(null)
+            } else {
+                throw IllegalStateException("Expected error, but got success")
+            }
+        }
+        setNavigationRoutes(routes, legIndex, callback)
+    }
+}
+
+suspend fun MapboxNavigation.setNavigationRoutesAndWaitForAlternativesUpdate(
+    routes: List<NavigationRoute>
+) =
+    withTimeout(MAX_TIME_TO_UPDATE_ROUTE) {
+        coroutineScope {
+            launch {
+                waitForAlternativeRoute()
+            }
+            setNavigationRoutes(routes)
+        }
```

Comment:
`setNavigationRoutes` is not `suspend`, can we simplify?
```suggestion
        setNavigationRoutes(routes)
        waitForAlternativeRoute()
```